### PR TITLE
Migrate PluginConfigInstanceLister

### LINKED ISSUE #133: [Builder] Create/Refactor PCI Parsing Logic
Implement the changes required to support parsing the v2 PCIs

Parent Issue: #113

### PR DESCRIPTION
**Pull request by engechas**
### Description
This PR migrates the logic necessary to list the available plugins. A user would invoke this code path by running `opensearch-benchmark list opensearch-plugins`, and receive an output such as:
```
Available OpenSearch plugins:

Name                     Configuration    Format Version
-----------------------  ---------------  ----------------
analysis-icu                              v1
analysis-kuromoji                         v1
analysis-phonetic                         v1
analysis-smartcn                          v1
analysis-stempel                          v1
analysis-ukrainian                        v1
discovery-azure-classic                   v1
discovery-ec2                             v1
discovery-file                            v1
discovery-gce                             v1
ingest-attachment                         v1
lang-javascript                           v1
lang-python                               v1
mapper-attachments                        v1
mapper-murmur3                            v1
mapper-size                               v1
repository-azure                          v1
repository-gcs                            v1
repository-hdfs                           v1
repository-s3                             v1
store-smb                                 v1
transport-nio                             v1
transport-nio            transport        v1
transport-nio            http             v1
```
 
I tested this code by wiring it into the list method to produce the above and via unit tests.

### Issues Resolved
#133 
 
### Check List
- [x] New functionality includes testing
  - [x] All unit and integration tests pass
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: osbenchmark/builder/configs/utils/config_path_resolver.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import os
+
+from osbenchmark.exceptions import SystemSetupError
+from osbenchmark.utils.repo import BenchmarkRepository
+
+
+class ConfigPathResolver:
```
[the remark this answers is not in the document]
Comment (engechas) [reply osbenchmark/builder/configs/utils/config_path_resolver.py:7]: It is used for both provision and plugin configs

File: osbenchmark/builder/configs/utils/config_path_resolver.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            raise SystemSetupError(f"Path {root_path} for {config_type} does not exist.")
+
+        return root_path
+
+    def _get_config_root_path(self):
+        root_path = self.cfg.opts("builder", "provision_config.path", mandatory=False)
+        if root_path:
+            return root_path
+        else:
+            distribution_version = self.cfg.opts("builder", "distribution.version", mandatory=False)
```
Comment (IanHoang) [inline osbenchmark/builder/configs/utils/config_path_resolver.py:24]: Nit: Seems like terms such as "builder", "provision_configs" are repeated in a variety of places. Might be better to place them in an enum like what you did with "opensearch" references.

https://github.com/opensearch-project/opensearch-benchmark/pull/183

File: osbenchmark/builder/models/config_instance_types.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from enum import Enum
+
+
```
Comment (IanHoang) [inline osbenchmark/builder/models/config_instance_types.py:3]: I think it's worth adding doc string to this Enum explaining its usage and significance

File: tests/builder/configs/utils/config_path_resolver_test.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    @mock.patch('osbenchmark.utils.git.fetch')
+    @mock.patch('osbenchmark.utils.repo.BenchmarkRepository')
+    @mock.patch('osbenchmark.utils.repo.BenchmarkRepository.setRepository')
+    @mock.patch('os.path.exists')
+    def test_provision_config_path_not_defined(self, path_exists, set_repo, benchmark_repo, git_fetch):
+        path_exists.return_value = True
+
+        # opts("builder", "provision_config.path"), opts("builder", "distribution.version"), opts("builder", "repository.name"),
+        # opts("builder", "repository.revision"), opts("system", "offline.mode"), opts("provision_configs", "%s.dir" % repo_name),
+        # opts("node", "root.dir"), opts("builder", "provision_config.repository.dir")
```
Comment (IanHoang) [inline tests/builder/configs/utils/config_path_resolver_test.py:34]: Is this intended to be temporarily commented out or can it be removed?

File: tests/builder/configs/utils/config_path_resolver_test.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        self.cfg.opts.side_effect = [None, "1.0", "fake-repo", "fake-revision", False, "fake-repo.dir", "/root_dir", "repo_dir"]
+
+        config_path = self.config_path_resolver.resolve_config_path(self.config_type, self.config_format_version)
+        self.assertEqual(config_path, "/root_dir/repo_dir/fake-repo/red/v36")
+
+    @mock.patch('os.path.exists')
+    def test_provision_config_path_does_not_exist(self, path_exists):
+        path_exists.return_value = False
+        # opts("builder", "provision_config.path")
```
Comment (IanHoang) [inline tests/builder/configs/utils/config_path_resolver_test.py:43]: Same as above: Is this also intended to be temporarily commented out or can it be removed?

**Review by IanHoang — APPROVED** [review]
Added a few comments but otherwise LGTM

File: osbenchmark/builder/models/config_instance_types.py (hunk 3 above)
Comment (engechas) [reply osbenchmark/builder/models/config_instance_types.py:3]: Will do

File: tests/builder/configs/utils/config_path_resolver_test.py (hunk 4 above)
Comment (engechas) [reply tests/builder/configs/utils/config_path_resolver_test.py:34]: The idea was to show the corresponding calls for the return values populated via `side_effect` on line 35

**Review by achitojha — APPROVED** [review]

**Pull request merged**